superset/common/query_context_processor.py

```diff
@@ -245,11 +246,21 @@ def normalize_df(self, df: pd.DataFrame, query_object: QueryObject) -> pd.DataFr
             if dttm_col:
                 timestamp_format = dttm_col.python_date_format
 
+        base_axis_label = None
+        if (
+            (label := get_base_axis_label(query_object.columns))
+            and query_object.datasource is not None
+            and (base_column := query_object.datasource.get_column(label))
+            and base_column.is_dttm
+        ):
+            base_axis_label = label
+
         normalize_dttm_col(
             df=df,
             timestamp_format=timestamp_format,
             offset=datasource.offset,
             time_shift=query_object.time_shift,
+            dttm_alias=base_axis_label,
```

Comment:
In reference to my comment about changing `get_base_axis_label` to plural form, could we loop through all `base_axis_labels` here to future proof for multi-dimensional charts?

Reply:
nice! it's also treating multiple DateTime columns. I have addressed this.